### ISSUE TITLE
Improvements to influxd upgrade command

### ISSUE THREAD

**Issue by stuartcarnie** [influxdata/influxdb#19985]
## What

In order to improve the upgrade experience for users moving from 1.x to 2.x, the following improvements should be added to the `influxd upgrade` command:

- [x] If the `meta.db` contains existing continuous queries, these should be written out to a separate file, continuous_queries.txt. This path should be available as a flag that can be changed, with a default value along side the `upgrade.log`
- [x] If the influxdb 1.x configuration file has disabled HTTP authentication and has users, this should be logged as a **warning**. 

    ```
    # Determines whether HTTP authentication is enabled.
    # auth-enabled = false
    ```

### Disabled HTTP authentication

If this feature is disabled in InfluxDB 1.x, clients sending requests with credentials are ignored by `influxd`. This may result in failures after upgrading for the following reasons:

* If the 1.x instance has users and 1.x clients, such as telegraf, are sending credentials, they will be silently ignored by InfluxDB 1.x
* Once upgraded to 2.x, those clients' requests will now be authorised by `influxd` 2.x and fail with a `401 Unauthorized` response if the credentials do not match.

**NOTE:** We may wish to alter the upgrade process in future, such that if HTTP auth is disabled, the users are not copied to 2.x.

**Comment by danxmoran** [comment]
I would also like to add:
- [ ] Move as many input validations as possible to the beginning of execution (i.e. input paths pointing to nonexistent files, output paths pointing to existing files)

**Comment by danxmoran** [comment]
Closed in #20012 #20036 #20039
